base/applications/rapps/applicationinfo.cpp

```diff
@@ -0,0 +1,626 @@
+    RichEdit->InsertText(szLoadedTextAvailability, NULL);
+    RichEdit->InsertText(szLangInfo, CFE_ITALIC);
+}
+
+VOID
+CAvailableApplicationInfo::RetrieveLanguages()
+{
+    m_LanguagesLoaded = true;
+
+    CStringW szBuffer;
+    if (!m_Parser->GetString(L"Languages", szBuffer))
+    {
+        return;
+    }
+
+    // Parse parameter string
+    int iIndex = 0;
+    while (true)
+    {
+        CStringW szLocale = szBuffer.Tokenize(L"|", iIndex);
+        if (szLocale.IsEmpty())
+            break;
+
+        szLocale = L"0x" + szLocale;
+
+        INT iLCID;
+        if (StrToIntExW(szLocale, STIF_SUPPORT_HEX, &iLCID))
+        {
+            m_LanguageLCIDs.Add(static_cast<LCID>(iLCID));
+        }
+    }
+}
+
+
+BOOL
+CAvailableApplicationInfo::Valid() const
+{
+    return !szDisplayName.IsEmpty() && !m_szUrlDownload.IsEmpty();
+}
+
+BOOL
+CAvailableApplicationInfo::CanModify()
+{
+    return FALSE;
+}
+
+BOOL
+CAvailableApplicationInfo::RetrieveIcon(CStringW &Path) const
+{
+    Path = szDisplayIcon;
+    return !Path.IsEmpty();
+}
+
+#define MAX_SCRNSHOT_NUM 16
+BOOL
+CAvailableApplicationInfo::RetrieveScreenshot(CStringW &Path)
+{
+    if (!m_ScrnshotRetrieved)
+    {
+        static_assert(MAX_SCRNSHOT_NUM < 10000, "MAX_SCRNSHOT_NUM is too big");
+        for (int i = 0; i < MAX_SCRNSHOT_NUM; i++)
+        {
+            CStringW ScrnshotField;
+            ScrnshotField.Format(L"Screenshot%d", i + 1);
+            CStringW ScrnshotLocation;
+            if (!m_Parser->GetString(ScrnshotField, ScrnshotLocation))
+            {
+                // We stop at the first screenshot not found,
+                // so screenshots _have_ to be consecutive
+                break;
+            }
+
+            if (PathIsURLW(ScrnshotLocation.GetString()))
+            {
+                m_szScrnshotLocation.Add(ScrnshotLocation);
+            }
+        }
+        m_ScrnshotRetrieved = true;
+    }
+
+    if (m_szScrnshotLocation.GetSize() > 0)
+    {
+        Path = m_szScrnshotLocation[0];
+    }
+
+    return !Path.IsEmpty();
+}
+
+VOID
+CAvailableApplicationInfo::GetDownloadInfo(CStringW &Url, CStringW &Sha1, ULONG &SizeInBytes) const
+{
+    Url = m_szUrlDownload;
+    m_Parser->GetString(L"SHA1", Sha1);
+    INT iSizeBytes;
+
+    if (m_Parser->GetInt(L"SizeBytes", iSizeBytes))
+    {
+        SizeInBytes = (ULONG)iSizeBytes;
+    }
+    else
+    {
+        SizeInBytes = 0;
+    }
+}
+
+VOID
+CAvailableApplicationInfo::GetDisplayInfo(CStringW &License, CStringW &Size, CStringW &UrlSite, CStringW &UrlDownload)
+{
+    License = LicenseString();
+    Size = m_szSize;
+    UrlSite = m_szUrlSite;
+    UrlDownload = m_szUrlDownload;
+}
+
+BOOL
+CAvailableApplicationInfo::UninstallApplication(BOOL bModify)
+{
+    ATLASSERT(FALSE && "Should not be called");
+    return FALSE;
+}
+
+
+
+CInstalledApplicationInfo::CInstalledApplicationInfo(HKEY Key, const CStringW &KeyName, AppsCategories Category, int KeyIndex)
```

Comment:
sometimes you use 1 empty lines and sometimes 2 to separate functions. I ignored that as I found it to be a minor nitpick, Still noting that the inconsistency makes it harder for other devs to follow any style. **But when its 3 lines like here, then my tolerance line is crossed and I do issue a comment. ;-)**